src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -50,11 +53,21 @@ public Publish(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFac
         public async Task<IEnumerable<PublishResult>> ProcessAsync(string folderPath)
         {
             this.publishResults.Clear();
+
             if (!this.IsValidInput(folderPath))
             {
                 return this.publishResults;
             }
 
+            var result = this.validateProduct.IsProductAndFolderValid(folderPath);
```

Comment:
Extract folder validation to higher level. In order to validate it even before publishing.

Reply:
Folder validation is separate now.